Crash Report: android - v1.5.8 (android)

## Crash Report

**Version:** 1.5.8
**Platform:** android
**Device:** android localhost


### Crash Log
```
=== JAVA CRASH REPORT ===
Time: 2026-03-20 16:23:08
Thread: QtBluetoothLEHandlerThread
Exception: android.os.DeadSystemRuntimeException
Message: android.os.DeadSystemException

=== STACK TRACE ===
android.os.DeadSystemRuntimeException: android.os.DeadSystemException
	at android.bluetooth.BluetoothGatt.writeCharacteristic(BluetoothGatt.java:1501)
	at org.qtproject.qt.android.bluetooth.QtBluetoothLE.executeWriteJob(QtBluetoothLE.java:1699)
	at org.qtproject.qt.android.bluetooth.QtBluetoothLE.performNextIO(QtBluetoothLE.java:1554)
	at org.qtproject.qt.android.bluetooth.QtBluetoothLE.access$1600(QtBluetoothLE.java:42)
	at org.qtproject.qt.android.bluetooth.QtBluetoothLE$3.run(QtBluetoothLE.java:1489)
	at android.os.Handler.handleCallback(Handler.java:958)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:205)
	at android.os.Looper.loop(Looper.java:294)
	at android.os.HandlerThread.run(HandlerThread.java:67)
Caused by: android.os.DeadSystemException
	... 10 more

=== DEVICE INFO ===
Android: 14 (API 34)
Device: Teclast M50Mini

```

### Debug Log (last lines)
```
[ 755.088] DEBUG WeatherManager: Got sun times for 4 days
[ 765.073] WARN  BLEManager: Scale connection timeout - not found
[ 774.930] DEBUG [BLE QtTransport] !!! CONTROLLER ERROR: ConnectionError !!!
[ 774.932] WARN  [BLE AcaiaScale] Transport error: !!! CONTROLLER ERROR: ConnectionError !!!
[ 774.933] DEBUG [BLE QtTransport] >>> Controller state changed: Unconnected
[ 775.073] DEBUG Scale reconnect: attempt 2 of 3
[ 775.074] DEBUG BLEManager: Direct wake - connecting to "LUNAR-57CFE9" at "60:8A:10:57:CF:E9"
[ 775.107] DEBUG [BLE QtTransport] connectToDevice() called for LUNAR-57CFE9 (60:8A:10:57:CF:E9). controller=yes state=0
[ 775.107] DEBUG [BLE QtTransport] Cleaning up previous controller
[ 775.108] DEBUG [BLE QtTransport] Connecting to LUNAR-57CFE9 (60:8A:10:57:CF:E9)
[ 775.116] DEBUG [BLE QtTransport] Calling connectToDevice on controller...
[ 775.137] DEBUG [BLE QtTransport] >>> Controller state changed: Connecting
[ 781.075] DEBUG BatteryManager: battery= 100 % mode= On(55-65%) charger= OFF discharging= true status= DISCHARGING plugged= UNPLUGGED
[ 781.083] DEBUG [Memory] QObject deltas: -664 QQuickAccessibleAttached, -309 QQuickRectangle, -295 QQuickText, -239 QQuickLayoutAttached, -224 QQmlComponent
[ 781.084] DEBUG [Memory] RSS: 100.2 MB  Java heap: 4.3 / 256.0 MB  QObjects: 1479  peak: 121.7 MB
[ 788.005] DEBUG [Screensaver] Video transition #2 RSS:100.2 MB
[ 795.074] WARN  BLEManager: Scale connection timeout - not found
[ 805.138] DEBUG [BLE QtTransport] !!! CONTROLLER ERROR: ConnectionError !!!
[ 805.140] WARN  [BLE AcaiaScale] Transport error: !!! CONTROLLER ERROR: ConnectionError !!!
[ 805.140] DEBUG [BLE QtTransport] >>> Controller state changed: Unconnected
[ 822.687] DEBUG [Screensaver] Video transition #3 RSS:100.2 MB
[ 835.074] DEBUG Scale reconnect: attempt 3 of 3
[ 835.075] DEBUG BLEManager: Direct wake - connecting to "LUNAR-57CFE9" at "60:8A:10:57:CF:E9"
[ 835.109] DEBUG [BLE QtTransport] connectToDevice() called for LUNAR-57CFE9 (60:8A:10:57:CF:E9). controller=yes state=0
[ 835.109] DEBUG [BLE QtTransport] Cleaning up previous controller
[ 835.110] DEBUG [BLE QtTransport] Connecting to LUNAR-57CFE9 (60:8A:10:57:CF:E9)
[ 835.119] DEBUG [BLE QtTransport] Calling connectToDevice on controller...
[ 835.144] DEBUG Scale reconnect: retries exhausted, waiting for manual reconnect or app resume
[ 835.145] DEBUG [BLE QtTransport] >>> Controller state changed: Connecting
[ 841.084] DEBUG [Memory] RSS: 99.7 MB  Java heap: 4.9 / 256.0 MB  QObjects: 1479  peak: 121.7 MB
[ 855.073] WARN  BLEManager: Scale connection timeout - not found
[ 865.143] DEBUG [BLE QtTransport] !!! CONTROLLER ERROR: ConnectionError !!!
[ 865.143] WARN  [BLE AcaiaScale] Transport error: !!! CONTROLLER ERROR: ConnectionError !!!
[ 865.143] DEBUG [BLE QtTransport] >>> Controller state changed: Unconnected
[ 868.926] DEBUG [Screensaver] Video transition #4 RSS:99.7 MB
[ 901.087] DEBUG [Memory] RSS: 99.0 MB  Java heap: 5.3 / 256.0 MB  QObjects: 1479  peak: 121.7 MB
[ 901.640] WARN  makeCurrent(): no EGLSurface, likely Surface destroyed by Android.
[ 901.641] WARN  QRhiGles2: Failed to make context current. Expect bad things to happen.
[ 901.642] WARN  Failed to start frame
[ 901.677] DEBUG App suspended - sleeping scale (DE1 stays awake)
[ 901.678] DEBUG BatteryManager: Ensuring charger is ON (app exit/suspend safety)
[ 901.678] DEBUG Application exiting - shutting down devices
[ 901.679] DEBUG Sending DE1 to sleep on app exit
[ 901.680] DEBUG Waiting for BLE queue to drain before exit...
[ 903.581] WARN  BLE queue drain timed out after 2000 ms — sleep command may not have been delivered.
[ 903.582] DEBUG BatteryManager: Ensuring charger is ON (app exit/suspend safety)
[ 903.582] DEBUG AccessibilityManager shutting down

========== SESSION START: 2026-03-20T16:24:54 ==========
[   0.077] DEBUG App started - version 1.5.8 build 3176 (release) built Mar 19 2026 19:01:33 at "2026-03-20T16:24:55"
```

---
*Auto-reported by Decenza DE1 app*

--- a/android/src/io/github/kulitorum/decenza_de1/DecenzaActivity.java
+++ b/android/src/io/github/kulitorum/decenza_de1/DecenzaActivity.java
@@ -64,10 +64,28 @@ public class DecenzaActivity extends QtActivity {
         super.onDestroy();
     }
 
+    private static boolean isDeadSystemException(Throwable t) {
+        while (t != null) {
+            if (t.getClass().getName().contains("DeadSystem")) {
+                return true;
+            }
+            t = t.getCause();
+        }
+        return false;
+    }
+
     private void installJavaCrashHandler() {
         defaultHandler = Thread.getDefaultUncaughtExceptionHandler();
 
         Thread.setDefaultUncaughtExceptionHandler((thread, throwable) -> {
+            // DeadSystemException means Android's system_server crashed or the device
+            // is rebooting. Nothing we can do — skip the crash report to avoid noise.
+            if (isDeadSystemException(throwable)) {
+                Log.w(TAG, "DeadSystemException on thread " + thread.getName()
+                        + " — Android system is dying, suppressing crash report");
+                return;
+            }
+
             try {
                 // Get crash log path (same as C++ crash handler)
                 File filesDir = getFilesDir();

